src/bindings/python/src/openvino/frontend/pytorch/gptq.py

```diff
@@ -0,0 +1,120 @@
+
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# flake8: noqa
+# mypy: ignore-errors
+
+import torch
+from functools import partial
+
+# Wraps a single tensor to a module to prevent it from jit.freezing
+# It depends on a tensor dtype whether it will be preserved from freezing. Refer to the decoder code to learn which types will be preserved.
+class KeepWeight(torch.nn.Module):
+
+    def __init__(self, weight):
+        super().__init__()
+        self.weight = torch.nn.Parameter(weight, requires_grad=False)
+
+    def forward(self):
+        return self.weight
+
+
+# Produces a pattern that can be captured later and represented as a single u4 constant node
+def decompression_pattern(weights):
+    mask = torch.tensor(15, dtype=torch.uint8).to(weights.device)
+    return torch.stack((torch.bitwise_and(weights, mask), torch.bitwise_right_shift(weights, 4)), dim=-1)
+
+
+def patched_forward(self, *args, **kwargs):
+    if hasattr(self, "_hf_hook"):
+        args, kwargs = self._hf_hook.pre_forward(self, *args, **kwargs)
+
+    dtype = x.dtype
+    x = args[0]
+    outshape = x.shape[:-1] + (self.width,)
+    x = x.view(-1, x.shape[-1])
+    groups = self.qzeros.shape[0]
+    height = self.qweight.shape[0]
+
+    unpacked_weights = decompression_pattern(
+        self._openvino_u4_compression_submodule_qweights()).contiguous().view(height, -1, 8)
+    unpacked_weights = torch.transpose(
+        unpacked_weights, 1, 2).contiguous().view(-1, self.group_size, self.width)
+    unpacked_zp = decompression_pattern(
+        self._openvino_u4_compression_submodule_qzeros()).contiguous().view(groups, 1, -1)
+
+    unpacked_zp = unpacked_zp.to(dtype) + 1
+
+    unpacked_weights = (unpacked_weights.to(dtype) - unpacked_zp) * self.scales
+    unpacked_weights = unpacked_weights.view(-1, self.width)
+
+    out = x @ unpacked_weights
+
+    out = out.view(outshape)
+    if self.bias is not None:
+        out.add_(self.bias)
+
+    if hasattr(self, "_hf_hook"):
+        out = self._hf_hook.post_forward(self, out)
+    return out
+
+
+def patch_model(model):
+    for name, m in model.named_modules():
+        if hasattr(m, "_openvino_patch_orig_forward"):
+            # already patched, skipping
+            continue
+        # TODO: Check module type
+        if hasattr(m, "QUANT_TYPE") and m.QUANT_TYPE == "exllama":
+            if m.bits != 4:
+                raise ValueError(
+                    f"Exllama kernel supports only bits=4, requested bits={m.bits} for module {name}. Something is wrong in the model initialization.")
+
+            int4_in_int32 = 8
+            groups = m.qzeros.shape[0]
```

Comment:
```suggestion
            groups = m.qzeros.shape[0]
            m.width = m.qweight.shape[1]
```

Reply:
Why does it work without this change? Other models don't have correct `width` defined?